### PR TITLE
update LLM recipes

### PR DESCRIPTION
**Pull request by XuehaoSun**
## Type of Change

- Add int4 GPTQ & AutoRound recipes

- bug fix

## Description

- Add tiiuae/falcon-7b,  baichuan-inc/Baichuan-13B-Chat, baichuan-inc/Baichuan2-13B-Chat, baichuan-inc/Baichuan2-7B-Chat, bigscience/bloom-1b7, databricks/dolly-v2-12b, EleutherAI/gpt-neox-20b, mistralai/Mistral-7B-v0.1, THUDM/chatglm2-6b, THUDM/chatglm3-6b scripts

## Expected Behavior & Potential Risk

no

## How has this PR been tested?

local test

## Dependency Change?

no

### CODE REVIEW

File: examples/huggingface/pytorch/text-generation/quantization/llm_quantization_recipes.md [hunk 1]
```diff
@@ -24,19 +34,17 @@ git clone https://github.com/intel/intel-extension-for-transformers.git
...
 # install requirements
 cd examples/huggingface/pytorch/text-generation/quantization
 pip install -r requirements.txt
-pip install neural-compressor==2.4.1
-pip install transformers==4.32.0
-pip install torch==2.1.1+cpu --index-url https://download.pytorch.org/whl/cpu
-pip install intel-extension-for-pytorch==2.1.100
-pip uninstall lm_eval -y
-pip install git+https://github.com/EleutherAI/lm-evaluation-harness.git@cc9778fbe4fa1a709be2abed9deb6180fd40e7e2
+pip install git+https://github.com/intel/neural-compressor.git@6a013c561af17bbc925e36f5a797cae55dad20e7
```
Comment (chensuyue) [inline examples/huggingface/pytorch/text-generation/quantization/llm_quantization_recipes.md:44]: use pip install neural-compressor==2.5

File: examples/huggingface/pytorch/text-generation/quantization/llm_quantization_recipes.md [hunk 2]
```diff
@@ -6,15 +6,25 @@ The scripts [run_generation.py](./run_generation.py) provide two quantization ap
...
+| :-----------------------------------------------------------------: |
+|             [EleutherAI/gpt-j-6b](#eleutheraigpt-j-6b)              |
+|                [facebook/opt-1.3b](#facebookopt-13b)                |
+|                [facebook/opt-30b](#facebookopt-30b)                 |
+|        [meta-llama/Llama-2-7b-hf](#meta-llamallama-2-7b-hf)         |
+|       [meta-llama/Llama-2-13b-hf](#meta-llamallama-2-13b-hf)        |
+|       [meta-llama/Llama-2-70b-hf](#meta-llamallama-2-70b-hf)        |
+|               [tiiuae/falcon-40b](#tiiuaefalcon-40b)                |
+|                [tiiuae/falcon-7b](#tiiuaefalcon-7b)                 |
+|  [baichuan-inc/Baichuan2-7B-Chat](#baichuan-inc/Baichuan2-7B-Chat)  |
```
Comment (chensuyue) [inline examples/huggingface/pytorch/text-generation/quantization/llm_quantization_recipes.md:19]: Do you mark specific transformers==4.33.3 for this model?

Comment (XuehaoSun) [reply examples/huggingface/pytorch/text-generation/quantization/llm_quantization_recipes.md:19]: > Do you mark specific transformers==4.33.3 for this model?

These models already support transformers==4.35.2

File: examples/huggingface/pytorch/text-generation/quantization/llm_quantization_recipes.md (hunk 1 above)
Comment (XuehaoSun) [reply examples/huggingface/pytorch/text-generation/quantization/llm_quantization_recipes.md]: > use pip install neural-compressor==2.5

fixed

**Review by chensuyue — APPROVED** [review]

**Pull request merged**